test/functional/find_tests.js

```diff
@@ -3121,4 +3122,46 @@ describe('Find', function() {
       });
     }
   });
+
+  it('should respect client-level read preference', {
+    metadata: { requires: { topology: ['replicaset'] } },
+
+    test: function(done) {
+      const config = this.configuration;
+      const client = config.newClient({}, { monitorCommands: true, readPreference: 'secondary' });
+
+      if (!config.usingUnifiedTopology()) {
+        this.skip();
+        return;
+      }
+
+      client.connect((err, client) => {
+        expect(err).to.not.exist;
+
+        let selectedServer;
+        const selectServerStub = sinon.stub(client.topology, 'selectServer').callsFake(function() {
```

Comment:
note to self: make it _much_ easier for us to test the type of selected server for an operation..